### ISSUE TITLE
TypeError from sphinx on the testing image with 3.10

### ISSUE THREAD

**Issue by skirpichev** [readthedocs/readthedocs.org#8652]
* Read the Docs project URL: https://readthedocs.org/projects/diofant/
* Build URL (if applicable): https://readthedocs.org/projects/diofant/builds/15210052/

Build crashed with the following traceback:
```
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/cmd/build.py", line 280, in build_main
    app.build(args.force_all, filenames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/application.py", line 343, in build
    self.builder.build_update()
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 293, in build_update
    self.build(to_build,
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 307, in build
    updated_docnames = set(self.read())
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 414, in read
    self._read_serial(docnames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 435, in _read_serial
    self.read_doc(docname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 475, in read_doc
    doctree = read_doc(self.app, self.env, self.env.doc2path(docname))
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/io.py", line 189, in read_doc
    pub.publish()
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/docutils/core.py", line 217, in publish
    self.document = self.reader.read(self.source, self.parser,
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/io.py", line 109, in read
    self.parse()
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/docutils/readers/__init__.py", line 78, in parse
    self.parser.parse(self.input, document)
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/parsers.py", line 101, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/docutils/parsers/rst/states.py", line 170, in run
    results = StateMachineWS.run(self, input_lines, input_offset,
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/docutils/statemachine.py", line 239, in run
    context, next_state, result = self.check_line(
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/docutils/statemachine.py", line 451, in check_line
    return method(match, context, next_state)
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/docutils/parsers/rst/states.py", line 2342, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/docutils/parsers/rst/states.py", line 2354, in explicit_construct
    return method(self, expmatch)
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/docutils/parsers/rst/states.py", line 2096, in directive
    return self.run_directive(
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/docutils/parsers/rst/states.py", line 2146, in run_directive
    result = directive_instance.run()
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/ext/autodoc/directive.py", line 162, in run
    documenter.generate(more_content=self.content)
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/ext/autodoc/__init__.py", line 983, in generate
    self.document_members(all_members)
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/ext/autodoc/__init__.py", line 843, in document_members
    classes = [cls for cls in self.documenters.values()
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/ext/autodoc/__init__.py", line 844, in <listcomp>
    if cls.can_document_member(member, mname, isattr, self)]
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/ext/autodoc/__init__.py", line 2066, in can_document_member
    return inspect.isNewType(member) and isattr
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/util/inspect.py", line 215, in isNewType
    return isinstance(obj, typing.NewType)
TypeError: isinstance() arg 2 must be a type, a tuple of types or a union

Exception occurred:
  File "/home/docs/checkouts/readthedocs.org/user_builds/diofant/envs/latest/lib/python3.10/site-packages/sphinx/util/inspect.py", line 215, in isNewType
    return isinstance(obj, typing.NewType)
TypeError: isinstance() arg 2 must be a type, a tuple of types or a union
The full traceback has been saved in /tmp/sphinx-err-ac6oxnh0.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```
Note, that the 3.10 docs build was successful in the project CI, see e.g. [this](https://github.com/diofant/diofant/actions/runs/1433430124).

**Comment by stsewd** [comment]
The error is https://github.com/sphinx-doc/sphinx/issues/9489. But looks like it should have been fixed on 4.2.

Something interesting from the logs

> created virtual environment CPython3.10.0.beta.4-64 in 4317ms

I think maybe that's the cause of the error. The stable version of python should be installed.

**Comment by skirpichev** [comment]
@stsewd, thanks - that works: https://readthedocs.org/projects/diofant/builds/15218818/
